### PR TITLE
feat: Update LandingPage2Template to include H1 tag

### LINKED ISSUE #11543: Update LandingPage2Template to include H1 tag
As part of the [Headings and Labels success criterion](https://cxwiki.sap.com/display/spar/Headings+and+Labels) for Screen Reader support, we need to have an `h1` tag in the DOM for the rotor narration to convey proper page hierarchy. 

Requirements: 
- This is not related to the `<title>` tag 
- This `<h1>` tag will be 'visually hidden' since our current design does not call for a Page title heading. 
- Styles can be used to offset the tab from the viewport to make it visually hidden
- Content of this `<h1>` tag can simply be "Homepage" (or match the page title)
- We already have `<h3>` tags for carousel titles that will be converted to `<h2>` in a separate ticket 

Dev Related Info (more investigation needed): 
- Make a new component `PageTitleComponent` based on `bread-crumb component
- Only need `h1` from the `bread-crumb component. no `<nav>` elements.
- Use `.cx-visually-hidden` class to hide it (CSS code provided in PR comments) 
- Make it configurable? (can be added to any page/content slot via Impex) 
- This can be added to other Landing Page templates as well? (it's a component that can be added to any content slot)

To Investigate: 
- List of pages that do not have a bread crumb (maybe in sample data?)

### PR DESCRIPTION
**Pull request by Pucek9**
closes #11543

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+
+@Component({
+  selector: 'cx-page-header',
+  templateUrl: './page-header.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PageHeaderComponent implements OnInit {
```
Comment (plabadie) [inline projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: Could the component benefit from a more self explanatory name? The component doesn't seem to be related to the page header or being a page header.

Not saying we should absolutely change the name, but let's discuss.

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PageHeaderComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (plabadie) [inline projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts]: There are no tests in this spec.  Please add coverage.

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  title$: Observable<string>;
+
+  constructor(
+    public component: CmsComponentData<CmsPageHeaderComponent>,
+    protected pageMetaService: PageMetaService
+  ) {}
+
+  ngOnInit(): void {
+    this.setTitle();
+  }
```
Comment (plabadie) [inline projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: I think there is a convention of moving away from using ngOnInit() for this sort off thing.  If we do move away from ngOnInit, we'd have to make sure it's not a breaking change.

Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: @plabadie paradoxically, kicking it from ngOnInit will be breaking changes😅 Please check BreadcrumbComponent changes. I removed this stuff from there, and add inheritance. It means that now its safe for current implementations

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts (hunk 2 above)
Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts]: TBH I didn't add any new logic here, I only divided BreadCrumbComponent to two smaller components. But sure, I can add it

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts (hunk 1 above)
Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: @plabadie what is your proposal? Would the name `PageTitleComponent` give it better?

File: projects/storefrontstyles/scss/components/content/navigation/_page-header.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+%cx-page-header {
+  position: absolute;
+  margin-top: -10000px;
+}
```
Comment (developpeurweb) [inline projects/storefrontstyles/scss/components/content/navigation/_page-header.scss]: In this case, only the element (not the component) needs to be visually hidden. E.g. The hidden H1 in the homepage will look something like this in the code: `<h1 class="cx-visually-hidden">Only available to Screen Reader users</h1>`

Therefore, please create a new class, name it `.cx-visually-hidden` and copy over the properties from Bootstrap `.sr-only` rule as per the screenshot below. Since we will drop Bootstrap at some point we need to create our own class. Imho, this new class should be **global**.

cc. @morganm58 please take note of the new **global** class request, let us know if any objections.

bootstrap Screen Shot 2021-03-30 at 5 26 44 PM

**Review by developpeurweb — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts (hunk 3 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: Agreed, let's play it safe and leave it as-is.

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts (hunk 1 above)
Comment (saad-mo) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts]: Discussed with Patrik and agreed to `PageTitleComponent`

File: projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.spec.ts]: Ah I understand!  It didn't have coverage to begin with.  Since it's separated out in it's own component and it results in an empty spec, if you can add some coverage, it would be great.

File: projects/storefrontstyles/scss/components/content/navigation/_page-header.scss (hunk 4 above)
Comment (saad-mo) [reply projects/storefrontstyles/scss/components/content/navigation/_page-header.scss]: I agree with this approach of Styles vs. negative margin originally proposed in the ticket. 

@Pucek9, I will update the ticket to to reflect this change.

Comment (Pucek9) [reply projects/storefrontstyles/scss/components/content/navigation/_page-header.scss]: > `<h1 class="cx-visually-hidden">Only available to Screen Reader users</h1>`

@developpeurweb are there any objections about add this class directly to component instead of h1? If it must be in h1 mandatory please let me know, I will change it quickly

Comment (developpeurweb) [reply projects/storefrontstyles/scss/components/content/navigation/_page-header.scss]: @Pucek9 h1 mandatory, please.

**Review by developpeurweb — APPROVED** [review]

**Review by plabadie — APPROVED** [review]

**Review by plabadie — APPROVED** [review]

**Review by saad-mo — APPROVED** [review]
Looks good!
